### DISCUSSION TITLE
Invoking artisan lambda - argv not populated

### DISCUSSION THREAD

**Discussion by FlorianRiquelme** [brefphp/bref#1560] [Support] [answered]
Hey everyone,

I'm struggling to execute artisan commands, on each invocation it just executes `artisan` without any parameter. I expected `argv` to have the arguments, but it doesn't which is likely my issue: 
```
[argv] => Array
(
[0] => /opt/bref/bootstrap.php
)
```

I'm using cdk to deploy this lambda:
```
import {ConsoleFunction, packagePhpCode} from "@bref.sh/constructs";

const artisanLambda = new ConsoleFunction(this, 'ArtisanLambda', {
            vpc: props.vpc,
            vpcSubnets: {
                subnetType: SubnetType.PRIVATE_ISOLATED,
            },
            handler: 'artisan',
            code: packagePhpCode(path.join(__dirname, '../../')),
            phpVersion: '8.2',
        });
```

I tried invoking it via cli like this:
```
aws lambda invoke \                                                                         
    --function-name myFunctionName \
    --region eu-central-1 \
    --cli-binary-format raw-in-base64-out \
    --payload '"-v"' \           
    response.json
```
And also tried the bref dashboard, both with the same result. 

Does someone have an idea how I can further debug this? Thanks :)

**Comment by mnapoli** [comment, marked as answer]
Could you check in the AWS console that the function correctly has the `php-82` layer, and then (*after*) the `php-82-console` layer?

**Reply by FlorianRiquelme** [reply]
Thanks for the quick reply!
That's it! I swapped the merge order manually and it works now.

Tried it with cdk:
```
layers: [
         LayerVersion.fromLayerVersionArn(this, 'PhpLayer', 'arn:aws:lambda:eu-central-1:534081306603:layer:php-82:37'),
         LayerVersion.fromLayerVersionArn(this, 'ConsoleLayer', 'arn:aws:lambda:eu-central-1:534081306603:layer:console:47'), 
]
```
And it still didnt work, but switching the layers did work. Seems counter intuitive that the last added layer will actually be the first one to be merged. Also the tests in the bref constructs repo [here](https://github.com/brefphp/constructs/blob/main/test/function/ConsoleFunction.test.ts#L18) won't catch it. 
I'll dig deeper after work, thank you for helping me out already. If you need any more information just ping me!

**Reply by mnapoli** [reply]
Oh wow that's so weird! So there's a bug in the base constructs right? Feel free to open a bug report, or even a PR if you have time!
